# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/android/nsWindow.cpp
# Commit: 670b6951b647
# Full Hash: 670b6951b647f11d1e4e60b026f36e26358578e2
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2020-11-14 09:46:25
# Description:
#   Bug 1676814 - Revert a2f0dae16c37 to stop crashing on fast fling etc. r=botond,geckoview-reviewers,snorp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96965
# ==============================================================================

diff -r cc937e3d332f -r 670b6951b647 widget/android/nsWindow.cpp
--- a/widget/android/nsWindow.cpp	Fri Nov 13 18:45:40 2020 +0200
+++ b/widget/android/nsWindow.cpp	Fri Nov 13 15:21:59 2020 +0000
@@ -360,9 +360,13 @@
         WheelDeltaAdjustmentStrategy::eNone);
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
+    int32_t ret =
+        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+            ? INPUT_RESULT_HANDLED
+            : INPUT_RESULT_HANDLED_CONTENT;
+
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
-      MOZ_ASSERT(result.mHandledResult, "Should have a valid APZHandledResult");
-      return ConvertAPZHandledResult(result.mHandledResult.value());
+      return ret;
     }
 
     PostInputEvent([input, result](nsWindow* window) {
@@ -374,9 +378,7 @@
       case nsEventStatus_eIgnore:
         return INPUT_RESULT_UNHANDLED;
       case nsEventStatus_eConsumeDoDefault:
-        return (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
-                   ? INPUT_RESULT_HANDLED
-                   : INPUT_RESULT_HANDLED_CONTENT;
+        return ret;
       default:
         MOZ_ASSERT_UNREACHABLE("Unexpected nsEventStatus");
         return INPUT_RESULT_UNHANDLED;
@@ -501,9 +503,13 @@
         nsWindow::GetEventTimeStamp(aTime), nsWindow::GetModifiers(aMetaState));
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
+    int32_t ret =
+        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+            ? INPUT_RESULT_HANDLED
+            : INPUT_RESULT_HANDLED_CONTENT;
+
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
-      MOZ_ASSERT(result.mHandledResult, "Should have a valid APZHandledResult");
-      return ConvertAPZHandledResult(result.mHandledResult.value());
+      return ret;
     }
 
     PostInputEvent([input, result](nsWindow* window) {
@@ -515,9 +521,7 @@
       case nsEventStatus_eIgnore:
         return INPUT_RESULT_UNHANDLED;
       case nsEventStatus_eConsumeDoDefault:
-        return (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
-                   ? INPUT_RESULT_HANDLED
-                   : INPUT_RESULT_HANDLED_CONTENT;
+        return ret;
       default:
         MOZ_ASSERT_UNREACHABLE("Unexpected nsEventStatus");
         return INPUT_RESULT_UNHANDLED;
@@ -702,10 +706,10 @@
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
       if (returnResult) {
-        MOZ_ASSERT(result.mHandledResult,
-                   "Should have a valid APZHandledResult");
         returnResult->Complete(java::sdk::Integer::ValueOf(
-            ConvertAPZHandledResult(result.mHandledResult.value())));
+            (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+                ? INPUT_RESULT_HANDLED
+                : INPUT_RESULT_HANDLED_CONTENT));
       }
       return;
     }
